drivers/flash/Kconfig.stm32

```diff
@@ -23,4 +23,13 @@ config SOC_FLASH_STM32_DEV_NAME
 	help
 	 Specify the device name for the flash driver.
 
+config SCO_FLASH_STM32FX_FORCE_SELECTION
```

Comment:
Why do we need this, why don't we just add something like this between lines 14 & 15:

select FLASH_PAGE_LAYOUT

Reply:
this is tips, so far I reworked code form STM32F4x.  It is only possible (AFAIK) method to limit force selection to STM32F4x. If I put this where you suggested it will be applied to all STM32.

Reply:
you can do:

   select FLASH_PAGE_LAYOUT if CONFIG_SOC_SERIES_STM32F4X

Reply:
As I remember It didn't work well.

Reply:
I got the syntax wrong, but there isn't any reason it shouldn't work (need to drop the CONFIG_):

select FLASH_PAGE_LAYOUT if SOC_SERIES_STM32F4X